scripts/run-docker-all.sh

```diff
@@ -10,11 +12,18 @@ that if built-in plugins are used, their resource usage is part of Conduit's usa
 
 EOF
 
-SLEEP_TIME=60
+SLEEP_TIME=300
 for w in workloads/*.sh; do
+  if [[ "$w" == *"helper"* ]]; then
+    continue
+  fi
+
   __dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
   source ${__dir}/run-docker-workload.sh $CONDUIT_IMAGE "$w"
 
+  # The sleep time here creates a "gap" in monitoring tools,
+  # which makes it clear when a workload ended
+  # and when the next one started.
```

Comment:
The monitoring tools I refer to above, are DataDog and/or CloudWatch, where we simply see graphs with resource usage. So likely nothing Conduit-specific will help there.

IMHO, 5 minutes isn't actually much. We have 7 workloads, each running for 30 minutes. We will likely add more workloads, and also have some tests running putting Conduit under load for a longer period of time. Also, the tests run in background. With that, I don't think that an addition 30 minutes matter for us, or the costs, but I can lower to maybe 2 minutes.

Reply:
I thought we only care about the metrics we collect through our metrics endpoint, why do we care about DataDog and/or CloudWatch? The pipeline related metrics are all labeled with the pipeline name, while resource usage can be correlated to the time when the workload was run (this can be found out if we use different pipeline names).

Let's take a step back - we are trying to separate workloads to make it clearer to see which workloads produced which metrics, correct? Even with a gap between workloads, we would still need to find out which workload was running at which time before we can make any conclusions. I see the use for a shorter idle duration to let the machine cool down and let GC and OS reclaim the memory, but I think separating workloads shouldn't be the reason to do this.

> We have 7 workloads, each running for 30 minutes.

I also think we may be overshooting the target here. I understand why we would run _some_ workloads for a longer time, to see how Conduit behaves over longer periods. But I don't see why _all_ workloads would need to run this long. Running benchmarks will now take 4 hours which is quite long already, on top we plan to run more workloads in the future and more tools (afaik the plan is to test other streaming tools besides Conduit). When we started to work on this I also thought that the benchmarks could be something that our users could run themselves to figure out how Conduit behaves on their machine, but I strongly doubt people will put in 4 hours of their time to run this. I know I wouldn't.

---

All that said, lowering the sleep from 5 to 2 minutes would shorten the duration of the benchmarks by 18 minutes. You are right that in the context of 4 hours that's not a lot so I won't insist on this. I still think we need to revisit the workloads at some point and make them more targeted so they produce exactly what we need.

Reply:
> why do we care about DataDog and/or CloudWatch?

It's because we have detailed metrics for the resource usage there, and some metrics we cannot get through Conduit/Prometheus (CPU usage for example).

> we are trying to separate workloads to make it clearer to see which workloads produced which metrics, correct?

True, but only for the DataDog and/or CloudWatch metrics. That's where we need a gap, so that's clear (visually) when a test ended and when a new one began. For the metrics we collect using our tool (which we print to a CSV file and which originate from Conduit's `metrics` endpoint), we have it easy, because each such metrics is timestamped and also labeled with the workload name.

Regarding test duration: agreed, there's no need to run all tests for the same time, especially not for 30 minutes. In fact, we actually agreed on 10 minutes. However, I did intentionally let each test run for longer this time for a few reasons.
1. One and the same duration applies to all tests
2. Sometimes I could reproduce that deadlock very quickly, sometimes it took longer.
3. I wanted to see GC in action

I'll change the default durations in a bit. :+1: